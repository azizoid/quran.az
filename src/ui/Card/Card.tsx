import { ReactNode } from 'react'

export type CardProps = {
  title: string | ReactNode
  titleClassName?: string | null
  contentClass?: string | null
  className?: string
  size?: 'small' | 'medium'
  children?: ReactNode
}

const cardSizes = {
  small: 'px-4',
  medium: 'px-7',
} as const

export const Card = ({
  title,
  titleClassName = '',
  className = '',
  size = 'medium',
  children,
}: CardProps) => (
  <div className="card">
    <div className={`card-title ${cardSizes[size]} ${titleClassName}`}>{title}</div>
    <div className={`card-content ${cardSizes[size]} ${className}`}>{children}</div>
  </div>
)
