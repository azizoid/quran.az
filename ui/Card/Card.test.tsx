import React from 'react'
import { render } from '@testing-library/react'

import { Card } from './Card'

test("Card snapshot", ()=>{
  const { container } = render(<Card title="Dummy Title" />)

  expect(container).toMatchSnapshot()
})