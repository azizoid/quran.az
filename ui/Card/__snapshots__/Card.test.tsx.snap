// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <div
      class="card-title px-7 "
    >
      Dummy Title
    </div>
    <div
      class="card-content px-7 "
    />
  </div>
</div>
`;
